mautrix_telegram/__main__.py

```diff
@@ -271,6 +284,55 @@ async def manhole_global_namespace(self, user_id: UserID) -> dict[str, Any]:
             "Puppet": Puppet,
         }
 
+    async def _notify_bridge_blocked(self, is_blocked: bool = True) -> None:
+        msg = self.config["bridge.limits.block_ends_notification"]
+        if is_blocked:
+            msg = self.config["bridge.limits.block_begins_notification"]
+            next_notification = self._last_blocking_notification + self.config["bridge.limits.block_notification_interval_seconds"]
+            # We're only checking if the block is active, since the block-end notification will not be resent and we want it ASAP
+            if next_notification > int(time()):
+                return
+            self._last_blocking_notification = int(time())
+
+        admins = list(map(lambda entry: entry[0],
+            filter(lambda entry: entry[1] == 'admin',
+                self.config["bridge.permissions"].items()
+            )
+        ))
+        if len(admins) == 0:
+            self.log.debug('No bridge admins to notify about the bridge being blocked')
+            return
+
+        self.log.debug(f'Notifying bridge admins ({",".join(admins)}) about bridge being blocked')
+
+        if not self._admin_rooms:
+            self.log.debug("Fetching admin rooms from the homeserver")
+            admin_rooms = {}
+            joined_rooms = await self.az.intent.get_joined_rooms()
+            for room_id in joined_rooms:
+                members = await self.az.intent.get_room_members(room_id)
+                if len(members) == 2: # a DM with someone
```

Comment:
That functionality (called a "notice room" exists in mautrix-signal, but not here unfortunately).

Good point on the edgecases though. We could count all the other events as well: it defaults to just `Membership.JOIN`, but we could also limit this search to rooms with only 2 members who are either `JOIN` or `INVITE`. That should exclude regular chats with other people (since there'd be joins/invites for them), and recreate rooms if an admin leaves a previous one (since they'll be neither a JOIN nor an INVITE then). Does that make sense?

Reply:
Yes, that sounds fine.
If I reject and invite to a room, do I still have the membership "invited"?

Reply:
Rejecting an invite gives you a "leave" status, so under that logic you'd get invited to a brand new room.

Reply:
Tuned in [ab7b4b1](https://github.com/vector-im/mautrix-telegram/pull/37/commits/ab7b4b19cc125c9248b10771cbf6fd4b3a49e769)